AmplifyTools/amplify-tools.sh

```diff
@@ -6,13 +6,121 @@
 # SPDX-License-Identifier: Apache-2.0
 
 set -e
-export PATH=$PATH:$(npm bin -g)
 
 if ! which node >/dev/null; then
   echo "warning: Node is not installed. Visit https://nodejs.org/en/download/ to install it"
   exit 1
-elif ! test -f ./amplifytools.xcconfig; then
-  npx amplify-app --platform ios
+fi
+
+export PATH=$PATH:$(npm bin -g)
+
+# Note the use of tail -1 is important here because when upgrading between versions
+# the first time that you run these commands, we have seen this variable take on the value of:
+# """
+# Scanning for plugins...
+# plugin scan successful
+# 4.21.0
+# """
+AMP_APP_VERSION_CURRENT=`npx -q amplify-app --version |tail -1`
+AMP_APP_VERSION_MINIMUM="2.17.1"
+AMP_APP_VERSION_INVALID=0
+
+AMP_CLI_VERSION_CURRENT=`npx -q amplify --version |tail -1`
+AMP_CLI_VERSION_MINIMUM="4.21.0"
+AMP_CLI_VERSION_INVALID=0
+
+
+STRIP_ESCAPE_RESULT=
+stripEscapeUtil() {
```

Comment:
Rather than using global vars as "return values", consider echoing the result. The caller would then invoke as `result=$(function "arg")`

Reply:
I have mixed feelings on this, and I think i'm going to pass on this for the reason that we have a number of other commands in this functions that can easily emit data to stdout/stderr.